Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -5046,7 +5046,11 @@ private void PrioritizeAndSchedule(LoggingContext loggingContext, IEnumerable<No
                                 }
                             }
 
-                            pipRuntimeInfo.Priority = (criticalPath < 0 || criticalPath > MaxInitialPipPriority) ? MaxInitialPipPriority : unchecked((int)criticalPath);
+                            int priorityIncrements = (int.MaxValue / (Process.MaxPriority + 1));
+                            int priorityBase = m_pipTable.GetPipPriority(pipId) * priorityIncrements;
+                            int criticalPathPriority = (criticalPath < 0 || criticalPath > MaxInitialPipPriority) ? MaxInitialPipPriority : unchecked((int)criticalPath);
+                            criticalPathPriority = Math.Min(criticalPathPriority, priorityIncrements - 1);
+                            pipRuntimeInfo.Priority = priorityBase + criticalPathPriority;
 

```

Comment:
I would recommend a different approach. Reserve the top bits (say 7 + 1 for the unused sign bit) for the priority and XOR that with the critical path priority (which should be adjusted so that its maximum value is int.MaxValue >> 7)

Reply:
That is basically what I am doing right?  except for instead of 128 I am using 100.

---
In reply to: [280869710](https://github.com/Microsoft/BuildXL/pull/239#discussion_r280869710) [](ancestors = 280869710)

Reply:
going to assume you meant OR

---
In reply to: [280885317](https://github.com/Microsoft/BuildXL/pull/239#discussion_r280885317) [](ancestors = 280885317,280869710)